bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/googletv/GoogleTVMessageParser.java

```diff
@@ -0,0 +1,334 @@
+        msg = msg.substring(2);
+        charArray = msg.toCharArray();
+
+        logger.trace("{} - Received GoogleTV message - Length: {} Message: {}", callback.getThingID(), len, msg);
+
+        callback.validMessageReceived();
+
+        try {
+            if (msg.startsWith(DELIMITER_1A)) {
+                logger.warn("{} - GoogleTV Error Message: {}", callback.getThingID(), msg);
+            } else if (msg.startsWith(DELIMITER_0A)) {
+                // First message on connection from GTV
+                //
+                // 0a 5b08 ff 041256 0a 11 534849454c4420416e64726f6964205456 12 06 4e5649444941 18 01 22 02 3131 2a
+                // ---------------------LEN-SHIELD Android TV--------------------LEN-NVIDIA---------LEN---LEN-Android
+                // 24 636f6d2e676f6f676c652e616e64726f69642e74762e72656d6f74652e73657276696365 32
+                // LEN-com.google.android.tv.remote.service
+                // 0d 352e322e343733323534313333
+                // LEN-5.2.473254133
+                //
+                // 0a 5308 ff 04124e 0a 0c 42524156494120344b204742 12 04 536f6e79 18 01 22 01 39 2a
+                // ---------------------LEN-BRAVIA 4K GB---------------LEN-Sony-------LEN---LEN-Android Version
+                // 24 636f6d2e676f6f676c652e616e64726f69642e74762e72656d6f74652e73657276696365 32
+                // 0d 352e322e343733323534313333
+                //
+                // 0a 5408 ff 04124f 0a 0a 4368726f6d6563617374 12 06 476f6f676c65 18 01 22 02 3132 2a
+                // ---------------------LEN-Chromecast-------------LEN-Google---------LEN---LEN-Android Version
+                // 24 636f6d2e676f6f676c652e616e64726f69642e74762e72656d6f74652e73657276696365 32
+                // 0d 352e322e343733323534313333
+                //
+                // 0a 5708 ff 041252 0a 0d 4368726f6d6563617374204844 12 06 476f6f676c65 18 01 22 02 3132 2a
+                // ---------------------LEN-Chromecast HD----------------LEN-Google---------LEN---LEN-Android Version
+                // 24 636f6d2e676f6f676c652e616e64726f69642e74762e72656d6f74652e73657276696365 32
+                // 0d352e322e343733323534313333
+
+                if (callback.getLoggedIn()) {
+                    logger.warn("{} - Unexpected Login Message: {}", callback.getThingID(), msg);
+                } else {
+                    callback.sendCommand(
+                            new GoogleTVCommand(GoogleTVRequest.encodeMessage(GoogleTVRequest.loginRequest(4))));
+                }
+
+                String st = "";
+                int length = 0;
+                StringBuffer preambleSb = new StringBuffer();
+                StringBuffer manufacturerSb = new StringBuffer();
+                StringBuffer modelSb = new StringBuffer();
+                StringBuffer androidVersionSb = new StringBuffer();
+                StringBuffer remoteServerSb = new StringBuffer();
+                StringBuffer remoteServerVersionSb = new StringBuffer();
```

Comment:
I think there's no need of the thread-safety overehead of a `StringBuffer` here and `StringBuilder`s can be used instead?